### PR TITLE
Update cookiecutter to new standard via cruft

### PR DESCRIPTION
**Pull request by Zeitsperre**
### Pull Request Checklist:
- [ ] This PR addresses an already opened issue (for bug fixes / features)
    - This PR fixes #xyz
- [ ] (If applicable) Documentation has been added / updated (for bug fixes / features).
- [ ] (If applicable) Tests have been added.
- [x] This PR does not seem to break the templates.
- [x] CHANGES.rst has been updated (with summary of main changes).
  - [x] Link to issue (:issue:`number`) and pull request (:pull:`number`) has been added.

### What kind of change does this PR introduce?

* Updates the cookiecutter with the newest conventions and workflows

### Does this PR introduce a breaking change?

Boilerplate documentation is largely unchanged. Workflows are now more a bit more elegant, including automatic labelling, warnings about unsafe changes to workflows, security-related changes, etc.

`actions-versions-updater.yml` has been replaced with `Dependabot` (it's just better).

There's support for evaluating the OpenSSF Scorecard (this can be disabled if we want).

Code formatting tools are now hard-pinned. These need to be kept in sync with changes from `pre-commit`. `Dependabot` should do this task automatically via Pull Requests.

Versioning scheme is now SemVer 2.0-compliant:
* If the version doesn't end in `-dev` or `-dev.##`, `$ bump-my-version bump patch` will be called. This will set the version at `X.Y.Z+1-dev`. Otherwise, `$ bump-my-version bump build` will be called. This is all automated by the `bump-version.yml`.

When the version is ready for a release, it's up to the maintainer to call the following:
* `$ bump-my-version bump release` (for a patch release; i.e. `1.2.0` → `1.2.1`) or
* `$ bump-my-version bump minor` then `$ bump-my-version bump release` (for a minor release; i.e. `1.2.0` → `1.3.0`)

There is now also support for Documentation locales (translations) via `.po` files. These can be staged with `$ make initialize-translations`, whereby `.po` files will be generated and French (or other) language strings can be added.
    * **I'm not sure how this will interact with existing translation code. Might be worth it to leave this deactivated**

### Other information:

https://github.com/Ouranosinc/cookiecutter-pypackage/pull/30

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -5,11 +5,13 @@ on:
     branches:
       - main
     paths-ignore:
+      - .cruft.json
       - CHANGES.rst
       - README.rst
       - pyproject.toml
       - setup.cfg
       - setup.py
+      - tests/test_xscen.py
```
Comment (aulemahal) [inline .github/workflows/main.yml:14]: Why ?

File: tox.ini [hunk 2]
```diff
@@ -2,11 +2,11 @@
 min_version = 4.0
 envlist =
     lint
-    py{39,310,311}-esmpy
+    py{39,310,31,312}-esmpy
```
Comment (aulemahal) [inline tox.ini:5]:
```suggestion
    py{39,310,311,312}-esmpy
```
I don't think we support Python 3.1 ;)

**Review by aulemahal — APPROVED** [review]
Looks good!

File: .github/workflows/main.yml (hunk 1 above)
Comment (Zeitsperre) [reply .github/workflows/main.yml:14]: There is one test that changes when we run `bump-my-version`. We shouldn't need to restart our tests when we bump the version, but maybe that's a good thing to do? I'm open to removing this.

File: tox.ini (hunk 2 above)
Comment (Zeitsperre) [reply tox.ini:5]: Oof. Now that's a fatfinger typo.

File: .github/workflows/main.yml (hunk 1 above)
Comment (Zeitsperre) [reply .github/workflows/main.yml:14]: @aulemahal 

image

Comment (aulemahal) [reply .github/workflows/main.yml:14]: Hmm I see!

I suggest removing either:
- the version number from the tested line
- the test file itself.

I don't see the point of that "smoke test" now that we have a minimal test suite. 
 @RondeauG  ?

Comment (RondeauG) [reply .github/workflows/main.yml:14]: I agree that we could remove that test if it complicates things!

Comment (Zeitsperre) [reply .github/workflows/main.yml:14]: Smoke tests removed!

**Pull request merged**